src/wallet/wallet.cpp

```diff
@@ -69,6 +69,8 @@ CFeeRate CWallet::fallbackFee = CFeeRate(DEFAULT_FALLBACK_FEE);
 
 const uint256 CMerkleTx::ABANDON_HASH(uint256S("0000000000000000000000000000000000000000000000000000000000000001"));
 
+CCoinControl g_coincontrol;
```

Comment:
What is the problem with it as global variable? It saves having to pass the variable all the way through to `GetAvailableCoins`. But if you think that's better I can change

Reply:
The problems with this as a global variable is it need to be reset when we don't need coin control feature before calling the function, which will make the function hard to use because the caller need to know there is a global variable that control function behavior, which is unexpected by other people if they don't know how that function was implemented.